### ISSUE TITLE
[Accessibility Bug] Section List: Headers are not announced by narrator

### ISSUE THREAD

**Issue by miczhongMSFT** [facebook/react-native#27694]
React Native version: https://github.com/microsoft/react-native/archive/v0.60.0-microsoft.13.tar.gz"

## Steps To Reproduce

1. Turn on operating system's narrator
2. Make sure list items are focus-able (Links, Buttons)
3. Navigate to a list item using keyboard 

Describe what you expected to happen:
Narrator should announce the section header for the first time a section item is focused on.

Snack, code example, screenshot, or link to a repository:
